### ISSUE TITLE
Idea for password strengh calculation

### ISSUE THREAD

**Issue by e-alfred** [fcturner/passwords#182]
Hello,

would it be a good idea to use something like https://github.com/dropbox/zxcvbn for password strength calculation? Just a thought about that.

**Comment by fcturner** [comment]
Great idea. I created my own algorithm. 

Time to professionalize things.

**Comment by fcturner** [comment]
Hm, I don't know if I like this. I *do* like the idea to improve my own algorithm by implementing something better, but at Dropbox (where they use this), they are convinced that `correct horse battery staple` is super safe because of its composition. Well, it might be a strong password, but that's primarily because of its length (bits of entropy). When a brute force attack only uses lower case characters for a start (without dictionary knowledge), this password will be cracked far more easily than a password containing other characters too. They focus on human memory; the possibility to remember a password. I wouldn't compensate security with human error. Just use passwords like `v^>nVd]FF}9432^2&yw[!E4Sb4Z+Rp` (just generated in my app). You must not remember them, because you shouldn't ever see them yourself! Hide them on the app and copy them to clipboard. Paste on the relevant website and nobody will ever know its composition.
Another pitfall: what if I don't hide my passwords? Or somebody sees me type it in? Don't you think it would be **much** secure to let them read `v^>nVd]FF}9432^2&yw[!E4Sb4Z+Rp` instead of  `correct horse battery staple`? They will remember it instantly! Now that's really unsafe.

On my app:
`zxcvbn` gets a `Weak (1)`
 `correct horse battery staple` gets a `Strong (52)`
`v^>nVd]FF}9432^2&yw[!E4Sb4Z+Rp` gets a `Strong (101)`

I'm actually thinking about dictionary reads to make the middle one have a `Moderate` strength at the most, the opposite of Dropbox's `zxcvbn`. Because I think it's less secure like I explained.

I will look for another solution.

**Comment by e-alfred** [comment]
Yes, these are pretty good arguments against it. And the password manager is indeed a tool so that you don't have to think about every password for every page/service you have an account for. ;-)

**Comment by fcturner** [comment]
Exactly :+1:

**Comment by Onlyodin** [comment]
There should probably be configuration options for the app that allows the user to define their own complexity thresholds, or at the very least be able to choose from some presets.

**Comment by fcturner** [comment]
That's a good suggestion!
